lib/authentication/auth_web.js

```diff
@@ -2,36 +2,45 @@
  * Copyright (c) 2015-2021 Snowflake Computing Inc. All rights reserved.
  */
 
-var util = require('../util');
-var rest = require('../global_config').rest;
+const util = require('../util');
+const rest = require('../global_config').rest;
 
-var net = require('net');
-var querystring = require('querystring');
-var URLUtil = require('./../../lib/url_util');
+const net = require('net');
+const querystring = require('querystring');
+const URLUtil = require('./../../lib/url_util');
+const Util = require('./../../lib/util');
 
 /**
  * Creates an external browser authenticator.
  *
  * @param {String} host
  * @param {module} webbrowser
  * @param {module} httpclient
+ * @param {module} browserActionTimeout
  *
  * @returns {Object}
  * @constructor
  */
-function auth_web(host, webbrowser, httpclient)
-{
-  var open = typeof webbrowser !== "undefined" ? webbrowser : require('open');
-  var axios = typeof httpclient !== "undefined" ? httpclient : require('axios');
+function auth_web(host, webbrowser, httpclient, browserActionTimeout) {
 
-  var host = host;
-  var port = rest.HTTPS_PORT;
-  var protocol = rest.HTTPS_PROTOCOL;
-  var proofKey;
-  var token;
-  var data;
+  if (!Util.exists(host)) {
```

Comment:
do we need those assertions? it's internal code and if host or timeout are missing then connection config will fail fast?

Reply:
I check to be sure that I have all needed parameters